gnat2goto/driver/gnat2goto_itypes.adb

```diff
@@ -171,4 +173,56 @@ package body Gnat2goto_Itypes is
       return Do_Type_Reference (Etype (N));
    end Do_Itype_Record_Subtype;
 
+   function Do_Modular_Integer_Subtype (N : Entity_Id) return Irep is
+      Modular_Type : constant Irep := Do_Type_Reference (Etype (N));
+      Followed_Mod_Type : constant Irep :=
+        Follow_Symbol_Type (Modular_Type, Global_Symbol_Table);
+
+      S_Range : constant Node_Id := Scalar_Range (N);
+      Lower_Bound : constant Node_Id := Low_Bound (S_Range);
+      Upper_Bound : constant Node_Id := High_Bound (S_Range);
+
+      Lower_Bound_Value : Integer;
+      Upper_Bound_Value : Integer;
+   begin
+
```

Comment:
might be worth asserting that `followed_mod_type` is actually an `unsigned` or `mod_type`